### ISSUE TITLE
User's episodes ratings retrieval timeout when rating filter is empty

### ISSUE THREAD

**Issue by defhead** [trakt/api-help#271]
Hello,

In user's ratings retrieval endpoint, `[GET] https://api.trakt.tv/users/<id>/ratings/<type>/<rating>`, the "rating" filter is optional, but in fact, without it the request goes to timeout.

Example:
```
 curl -v -H "Content-Type: application/json" \
-H "Authorization: Bearer <MY_BEARER>" \
-H "trakt-api-key: <MY_API_KEY>" \
-H "trakt-api-version: 2" \
https://api.trakt.tv/users/<my_username>/ratings/episodes/?page=1&limit=5
```

Adding the "rating" filter, `https://api.trakt.tv/users/<my_username>/ratings/episodes/8?page=1&limit=5`, resolves the issue.

On movies, seasons and shows ratings retrieval, there is no issues when this filter is empty.

Maybe it's because my account has a lot of watched episodes (~18K), I don't know.

**Comment by rectifyer** [comment]
Yeah, 18k ratings is the issue. If you provide the username I can troubleshoot further and maybe see why it's timing out even with the pagination params.

**Comment by defhead** [comment]
Username : eleon

I made some requests on this API just right now, between 04/05/2022 21:20 and 21:40 (GMT). Some of them went to timeout.

Especially when I don't put "page" and/or "limit" parameters. Those are optional, it seems they don't have default values on server side.

**Comment by eliehaykal** [comment]
Here is another user who informed me of similar timeout issues when retrieving ratings: phox00

**Comment by rectifyer** [comment]
There is no fix when trying to request this much data at once. You will need to implement pagination and step through the results if you actually need to have all ratings at once. In the future, we will enforce pagination for exactly this reason of not being to process all the data fast enough in a single result set.

If you have more questions, please post specific details on what you're trying to do with all ratings in your app and I might have some additional suggestions.

**Comment by defhead** [comment]
@rectifyer 
It was not the goal. I didn't need to request this much data at once. **The timeout was happening even with "limit" and "page" query params.**
Look at the examples of the original message. 

Since this issue was created 5 years ago, I don't know anymore if it's still behaving like that.
